### DISCUSSION TITLE
How to add battery to existing system ?

### DISCUSSION THREAD

**Discussion by FilipeAmadeuO** [hoylabs/OpenDTU-OnBattery#1368] [Q&A]
Hi. I have a solar panel system with:
- 6 Hyundai 485W solar panels
- 2 Hoymiles HM-1500
Connected to energy network through a secundary power board

In my main board i have a Tasmota Power meter + OpenDTU and everything is integrated into Home Assistant, so i know real time solar production and home usage.

Now, i am planning to add bateries, but they cannot be located near the inverters/solar panels because of space limitation.
Question: Is there any way to add batteries inside on the house that can be controlled by OpenDTU and/or Tasmota ?

My schematics is following:
![Painel solar](https://github.com/user-attachments/assets/5f2c8d35-f177-4fbe-9cc0-835fbee323f3)

**Comment by spcqike** [comment]
If you really want to use openDTU for that, your best chance is to use a dedicated system with huawei AC charger, battery and a third Hoymiles. 

But I wouldn’t use this. I would go with a Victron multiplus system.

**Reply by FilipeAmadeuO** [reply]
It´s not mandatory to use OpenDTU. My main goal is to have the batteries installed on other place than the microinverters and have zero injection...

**Reply by Manos1966** [reply]
Hi @spcqike 
good example we have here.
Is it possible to explain how you would setup the Victron Multiplus, which model you would use and whether it will work with the Shelly 3em? 
...and how much it would cost?

The alternative I see for the setup at hand would indeed be a HUAWEI 4850 BUT....
need to find one (80EUR if lucky, otherwise 150EUR)
get Adapter (30EUR)
find a way to put everything together in a box (50-80EUR)
maybe add some basic protection features (40EUR)
add one more Huawei (200EUR)
and live with the personal liability... (3000W system we are talking about), it is prerequisite that one knows what he is doing! 

I am impressed by how well my HUAWEI 4850 is working with OpenDTU-OnBattery 💯 
But if I would like to have a "Fire and Forget" System that I will put at a corner and forget about it, I would prefer a more "heavy duty" solution.


@FilipeAmadeuO how good is your German? 🙊 
look here #1268

**Reply by spcqike** [reply]
I guess I would start with a MultiPlus-II 48/5000/70-50 GX. As I would like to expand it in the coming years to a three phase system. 

You can use a Shelly 3EM and add it as smart meter to Venus OS
https://github.com/fabian-lauer/dbus-shelly-3em-smartmeter

So, I don’t know as I haven’t build it yet, but I guess you need about ~1.000€ for the multiplus and stuff (bus bar, wires, fuses,…). Your battery comes on top. 
so ~2.000€ for a 5kW single phase system with 5kWh battery (pylontech) 

but as of know it’s just an idea. I plan to build something like this in 2025. 

My main advantage: scalability (starting with 5kW single phase and expanding to 15kW 3-phase and starting with 5kWh battery and expanding later) and ESS. I want to power my server and network with the ESS output. And you could plug in your hoymiles to this AC port and charge the battery, even in a global power outage.

**Reply by spcqike** [reply]
Not to forget: only one multiplus neefs to be a GX. The others can be non GX version. (All can be non GX, but than you need an external controller) 

you can also use just two multiplus and put them in parallel. So like going from 5kW single phase to 10kW. 

You can also use 6 multiplus later on. Starting with 5kW, going to 15kW three phase and expanding to 30kW three phase. 

You can also use one or more easysolar, which are multiple with integrated smartsolar MPPT charger. 

Or use external smartsolar chargers. 

The smallest multiplus starts at 3kW. But 9kW is to little, so I’m going to start with 5kW devices :) the biggest ones have 15kW singlephase (or was it 10kW?)

**Reply by Manos1966** [reply]
@spcqike what does the GX option do to the Multiplus? (100EUR more expensive)?

Do I need to install also separate HW (Raspberry etc) to have a Venus OS running or, will the Multiplus (GX?) take care of that?

**Reply by spcqike** [reply]
well, i'm not to sure right now, as i just started reading about it :) but as far as i understood, you kind of need a one "head unit" to manage multiple multipus (use a three phase, use 0 grid export). this unit can be one of multiple. there are several external devices, with or without display, and, the multiplus with integrated GX. 

venus OS can run on a raspberry, or in a VM. i would use my proxmox host to run a VM. afaiu you need venus OS, to connect your Shelly 3EM as overall energy meter to your system. otherwise you can use victrons EM which uses victron (mod?)bus.

i also saw, that there are projects to use an esp32 as "shelly to VE.bus bridge" (https://github.com/PepeTheFroggie/Victron-VE.Bus---esp32).


as i said, i'm not quite sure about that. at the moment i think i need a "GX"-unit for my later goal of using 3 multiplus as a three-phase system (but, of course i could start with one non GX and buy a GX version when needed) and i need venus OS of any kind because i want to use my existing Shelly 3EM. if i would stick to a victron EM, i think i wouldn't need venus OS. and, maybe i don't need it with the linked esp32 project, as this is a hardware bridge.

i still need to read about all of that :)

**Comment by gitisgreat2023** [comment]
@FilipeAmadeuO
You basically have two options with their +ve and -ves.
1) Victron Multiplus solution
+ve: 
a) Biggest by far: its a complete solution, 48V charge/discharge, emergency capable
b) expandable, just add more Multiplus's
-ve: 
a) you need an electrician to install (I assume so, not sure though), I guess that costs about 500-800€.
b) efficiency: At low power the round-trip is about 56%, at 200W 73%, peaks at 85% at 1300W and then goes down to 80% again at 3000W. Especially with your relative low power solar (2.9kW) that's eating away quite some of your kWh generated.
c) thermal throttling, the 3kW model only delivers that at 25°C, it almost halves for all the models when going to 65°C. (see b), low efficiency will cause fast heating the multiplus)
2) Huawei charger, HM inverter
+ve: 
a) high efficiency overall
b) you can do it yourself
c) modular: charger and discharger are separate
-ve: it's can be messy, you have to take care of a good Huawei cooling (I managed it though) to have low temperate at 3kW charging.

If you're technically skilled and you're seeing it more of a hobby, go for 2) otherwise stick to 1).

I went for 2), currently I also have a small solar, only 4.58kWp. In hindsight I should have skipped Pylontech und go for the DIY directly. I'll sell my Pylontech 5000 (any interest? ;-)) as soon as my Xixiang box and EVE MB31 have arrived. The DIY box is modular and the slightly higher voltage is an +ve as well (15S Pylontech vs 16S DIY).
If we finally have a big PV I would have to think again hard to see what I would do. Maybe Victron, maybe not.

**Comment by Manos1966** [comment]
> efficiency: At low power the round-trip is about 56%, at 200W 73%,

Aoutsch, this hurts!

Many thanks @spcqike and @gitisgreat2023  this is valuable information.

As @gitisgreat2023 said, "If you're technically skilled and you're seeing it more of a hobby, go for 2) otherwise stick to 1)

I would like to know a system that requires "no interaction" (not even the Shelly 3em but Victron own module instead):
Here is why: My neighbours are 75years old. They would need a system that is reliable and does not require too much monitoring because one day, I may move out and no longer be in the same area.
For the time being I am in favor of a DEYE+GOBEL but this is a system that cost 4000-5000 EUR (too much money for an old pair and questionable amortisation within their lifetime).
